sycl/test/aie/ranged_buffer.cpp

```diff
@@ -0,0 +1,31 @@
+// REQUIRES: aie
+
+// RUN: %aie_clang %s -o %t.bin
+// RUN: %if_run_on_device %run_on_device %t.bin > %t.check 2>&1
+// RUN: %if_run_on_device FileCheck %s --input-file=%t.check
+
+#include "aie.hpp"
+#include <numeric>
+
+int main() {
+  constexpr std::size_t size = 10;
+  aie::device<1, 1> dev;
+  aie::buffer<int> buff(10);
+  std::iota(buff.begin(), buff.end(), 0);
+  for (int i = 0; i < size; i++)
+    assert(buff[i] == i);
+  aie::queue q(dev);
+  q.submit_uniform([&](auto& ht) {
+    aie::accessor acc = aie::buffer_range(ht, buff)
+                            .read_range(0, buff.size())
+                            .write_range(0, buff.size());
```

Comment:
Thinking more about the API, this does not look like the SYCL wording.
Would it be possible to just have this function in a normal accessor?
```suggestion
    auto acc = aie::accessor { ht, buff }
                            .read_range(0, buff.size())
                            .write_range(0, buff.size());
```
or
```suggestion
    aie::accessor acc { ht, buff };
    acc.read_range(0, buff.size()).write_range(0, buff.size());
```
?

Reply:
Yes its possible. but I don't think its much better then the current state. and it would require a big rewire of accessors on the host.